src/core/auth/authToken.ts

```diff
@@ -0,0 +1,30 @@
+import jwt, { JsonWebTokenError } from "jsonwebtoken"
+import configs from "../configs"
+import { AuthRole } from "../enums"
+import { UnauthorisedError } from "../http"
+
+const secret = configs.jwt.secret
+const issuer = configs.server.name
+const subject = "auth-token"
+const expiresIn = "12h"
+
+export interface IAuthPayload {
+	role: AuthRole
+	id: string
+}
+
+export function createAuthToken(payload: IAuthPayload) {
+	return jwt.sign(payload, secret, { issuer, expiresIn, subject })
+}
+
+export function validateAuthToken(token: string): IAuthPayload {
```

Comment:
@kunalgosrani  can you try to change all the functions to arrow Functions as per ES6?

Reply:
I prefer arrow function only in case of lexical scope.
Normal function looks clean to me.